x-pack/plugins/alerting/server/routes/snooze_rule.ts

```diff
@@ -0,0 +1,57 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import { IRouter } from 'kibana/server';
+import { schema } from '@kbn/config-schema';
+import { ILicenseState, AlertTypeDisabledError } from '../lib';
+import { RewriteRequestCase, verifyAccessAndContext } from './lib';
+import { AlertingRequestHandlerContext, BASE_ALERTING_API_PATH } from '../types';
+
+const paramSchema = schema.object({
+  id: schema.string(),
+});
+
+const bodySchema = schema.object({
+  snooze_end_time: schema.maybe(schema.string()),
+});
+
+const rewriteBodyReq: RewriteRequestCase<{
+  snoozeEndTime?: string;
+}> = ({ snooze_end_time: snoozeEndTime }) => ({
+  snoozeEndTime,
+});
+
+export const snoozeRuleRoute = (
+  router: IRouter<AlertingRequestHandlerContext>,
+  licenseState: ILicenseState
+) => {
+  router.post(
+    {
+      path: `${BASE_ALERTING_API_PATH}/rule/{id}/_snooze`,
+      validate: {
+        params: paramSchema,
+        body: bodySchema,
+      },
+    },
+    router.handleLegacyErrors(
+      verifyAccessAndContext(licenseState, async function (context, req, res) {
+        const rulesClient = context.alerting.getRulesClient();
+        const params = req.params;
+        const body = rewriteBodyReq(req.body);
+        try {
+          await rulesClient.snooze({ ...params, ...body });
```

Comment:
maybe we can add some validation to not set snooze_end_time in the past, what do you think?